tests/README.rst

```diff
@@ -0,0 +1,74 @@
+===========
+Don't Panic
+===========
+
+The Hypothesis test suite is large, but we've written these notes to help you
+out.  It's aimed at contributors (new and old!) who know they need to add tests
+*somewhere*, but aren't sure where - or maybe need some hints on what kinds of
+tests might be useful.  Others might just be interested in how a testing
+library tests itself!
+
+
+The very short version
+======================
+
+- To improve code coverage (eg because ``make check-coverage`` / the Travis
+  build is failing), go to ``cover/``
+- For longer / system / integration tests, look in ``nocover/``
+- For tests that require an optional dependency, look in the appropriate
+  directory
+
+In each case there will be further notes to help you.
+
+
+About each group of tests
+=========================
+
+Still here?  Here's a note on what to expect in each directory.
+
+``common/``
+    Useful shared testing code, including test setup and a few helper
+    functions in ``utils.py``.  Also read up on pytest features such as
+    ``mark.parametrize``, ``mark.skipif``, and ``raises``.
+
+``cover/``
+    The home of enough tests to get 100% branch coverage - you'll probably
+    spend most of your time here.  Ideally this would be as quick as possible
+    while remaining comprehensive.  More details inside.
+
+``datetime/``
+    *Tests for the deprecated* ``hypothesis.extra.datetime`` *module, which
+    depends on the ``pytz`` package.*
+
+``django/``
+    Tests for the Django extra.  Includes a toy application, to give us lots
+    of models to generate.
+
+``fakefactory/``
+    *Tests for the deprecated Faker extra.*
```

Comment:
I knew this was unstable, but I didn't realise it was deprecated? Nothing in the docs mentions that: http://hypothesis.readthedocs.io/en/latest/extras.html?highlight=faker#hypothesis-fakefactory

Reply:
You're right!  I've just been thinking of it as deprecated because we've given up on compatibility with new versions and recommend that people avoid using it.  (except via use of existing providers, which are probably rare due to the age of the versions we support)

@DRMacIver, would you be OK with a programmatic deprecation here and explicit docs to match?

Reply:
This is now done, right?

Reply:
Yep, in #1005